Assets/SEE/Game/SceneManipulation/SpatialMetrics.cs

```diff
@@ -0,0 +1,253 @@
+        public Bounds2D (float left, float right, float back, float front)
+        {
+            Left = left;
+            Right = right;
+            Back = back;
+            Front = front;
+        }
+
+        /// <summary>
+        /// Gets or sets the element at the specified direction.
+        /// </summary>
+        /// <param name="dir">The direction of the element to get or set.</param>
+        /// <returns>The element at the specified direction.</returns>
+        /// <exception cref="IndexOutOfRangeException">
+        /// Thrown when the direction is ambiguous or zero.
+        /// </exception>
+        public float this[Vector3 dir]
+        {
+            get {
+                if (dir == Vector3.left)
+                {
+                    return Left;
+                }
+                if (dir == Vector3.right)
+                {
+                    return Right;
+                }
+                if (dir == Vector3.forward)
+                {
+                    return Front;
+                }
+                if (dir == Vector3.back)
+                {
+                    return Back;
+                }
+                throw new IndexOutOfRangeException($"Given direction is not possible in {nameof(Bounds2D)}: {dir}");
+            }
+
+            set {
+                if (dir == Vector3.left)
+                {
+                    Left = value;
+                }
+                else if (dir == Vector3.right)
+                {
+                    Right = value;
+                }
+                else if (dir == Vector3.back)
+                {
+                    Back = value;
+                }
+                else if (dir == Vector3.forward)
+                {
+                    Front = value;
+                }
+                throw new IndexOutOfRangeException($"Given direction is not possible in {nameof(Bounds2D)}: {dir}");
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the element at the specified direction.
+        /// </summary>
+        /// <param name="dir">The direction of the element to get or set.</param>
+        /// <returns>The element at the specified direction.</returns>
+        /// <exception cref="IndexOutOfRangeException">
+        /// Thrown when the direction is <see cref="Direction2D.None"/>.
+        /// </exception>
+        public float this[Direction2D dir]
+        {
+            get {
+                switch (dir)
+                {
+                    case Direction2D.Left:
+                        return Left;
+                    case Direction2D.Right:
+                        return Right;
+                    case Direction2D.Back:
+                        return Back;
+                    case Direction2D.Front:
+                        return Front;
+                    default:
+                        throw new IndexOutOfRangeException($"Given direction is not possible in {nameof(Bounds2D)}: {dir}");
+                }
```

Comment:
This can be a switch expression:
```suggestion
                return dir switch
                {
                    Direction2D.Left => Left,
                    Direction2D.Right => Right,
                    Direction2D.Back => Back,
                    Direction2D.Front => Front,
                    _ => throw new IndexOutOfRangeException($"Given direction is not possible in {nameof(Bounds2D)}: {dir}")
                };
```